packages/manager/apps/telecom/src/app/telecom/telephony/service/contact/contact.controller.js

```diff
@@ -1,42 +1,160 @@
 import assign from 'lodash/assign';
 import find from 'lodash/find';
 import get from 'lodash/get';
-import map from 'lodash/map';
 import omit from 'lodash/omit';
 import pickBy from 'lodash/pickBy';
-import set from 'lodash/set';
 import some from 'lodash/some';
 import startsWith from 'lodash/startsWith';
-import toUpper from 'lodash/toUpper';
 import words from 'lodash/words';
+import head from 'lodash/head';
 
 import {
   DIRECTORY_INFO,
   DIRECTORY_WAY_NUMBER_EXTRA_ENUM,
+  LEGAL_FORM_ENUM,
+  AVAILABLE_FIELDS,
+  REGEX,
 } from './contact.constants';
 
-export default /* @ngInject */ function TelecomTelephonyServiceContactCtrl(
-  $state,
-  $stateParams,
-  $q,
-  $timeout,
-  $translate,
-  OvhApiTelephony,
-  TucToast,
-  TucToastError,
-  OvhApiXdsl,
-  tucTelephonyBulk,
-) {
-  const self = this;
-
-  function buildWayInfo(directory) {
+export default class TelecomTelephonyServiceContactCtrl {
+  /* @ngInject */
+  constructor(
+    $stateParams,
+    $q,
+    $timeout,
+    $translate,
+    OvhApiTelephony,
+    TucToast,
+    TucToastError,
+    tucTelephonyBulk,
+    TelecomTelephonyServiceContactService,
+  ) {
+    this.billingAccount = $stateParams.billingAccount;
+    this.serviceName = $stateParams.serviceName;
+    this.$q = $q;
+    this.$timeout = $timeout;
+    this.$translate = $translate;
+    this.OvhApiTelephony = OvhApiTelephony;
+    this.TucToast = TucToast;
+    this.TucToastError = TucToastError;
+    this.tucTelephonyBulk = tucTelephonyBulk;
+    this.TelecomTelephonyServiceContactService = TelecomTelephonyServiceContactService;
+    this.DIRECTORY_INFO = DIRECTORY_INFO;
+    this.DIRECTORY_WAY_NUMBER_EXTRA_ENUM = DIRECTORY_WAY_NUMBER_EXTRA_ENUM;
+    this.LEGAL_FORM_ENUM = LEGAL_FORM_ENUM;
+    this.AVAILABLE_FIELDS = AVAILABLE_FIELDS;
+    this.REGEX = REGEX;
+  }
+
+  $onInit() {
+    this.isLoading = true;
+
+    this.directoryCodes = null;
+    this.autoCompletePostCode = null;
+    this.autoCompleteCity = null;
+    this.autoCompleteStreetName = null;
+
+    return this.$q
+      .all({
+        infos: this.OvhApiTelephony.v6().schema().$promise,
+        directory: this.TelecomTelephonyServiceContactService.getDirectory(
+          this.billingAccount,
+          this.serviceName,
+        ),
+      })
+      .then(({ infos, directory }) => {
+        return this.$q.all({
+          infos,
+          directory,
+          autoCompletePostCode: this.postCodeAvailable(directory),
+        });
+      })
+      .then(({ infos, directory, autoCompletePostCode }) => {
+        this.autoCompletePostCode =
+          autoCompletePostCode?.length > 0
+            ? autoCompletePostCode.sort((a, b) => {
+                if (a < b) {
+                  return -1;
+                }
+                if (a > b) {
+                  return 1;
+                }
+                return 0;
+              })
```

Comment:
there is any raison to have your own comparison function ?
probably call to `autoCompletePostCode.sort()` is enough.

Reply:
It's applied